### PR TITLE
Improve unit tests

### PR DESCRIPTION
**Pull request by tstenner**
For an eventual py3-transition we need better unit test coverage.

What's planned / done:
- [X] Introduce a custom base class with better tempfile handling and asserts
- [ ] Convert asserts / tempfile creation to use the class
- [x] Convert asserts from `self.assert_` to more meaningful asserts (especially `self.assert_(isinstance(...))`
- [ ] test for unicode / bytes explicitly

### CODE REVIEW

File: tests/common.py [hunk 1]
```diff
@@ -26,24 +26,51 @@
...
 
+import unittest
+import shutil
 import tempfile
 import os
+import os.path
 
 
-class AssertFile:
+class BleachbitTestCase(unittest.TestCase):
```
Comment (az0) [inline tests/common.py]: It's a good idea to use inheritance here

File: tests/TestAction.py [hunk 2]
```diff
@@ -209,10 +202,7 @@ def execute_json(path, address):
 
     def test_process(self):
         """Unit test for process action"""
-        if 'nt' == os.name:
-            cmd = 'cmd.exe /c dir'
-        if 'posix' == os.name:
-            cmd = 'dir'
+        cmds = {'nt': 'cmd.exe /c dir', 'posix': 'dir'}
```
Comment (az0) [inline tests/TestAction.py]: This is a creative use of a dictionary. Nice.

File: tests/TestCommand.py [hunk 3]
```diff
@@ -21,70 +21,52 @@
...
-        self.assert_(os.path.getsize(path) > 0)
+        self.assertExists(path)
+        self.assertGreater(os.path.getsize(path), 0)
 
         # preview
         ret = cmd.execute(False).next()
-        self.assert_(os.path.exists(path))
-        self.assert_(os.path.getsize(path) > 0)
+        self.assertExists(path)
+        self.assertGreater(os.path.getsize(path), 0)
```
Comment (az0) [inline tests/TestCommand.py]: assertGreater is another benefit of Python 2.7+

File: tests/TestFileUtilities.py [hunk 4]
```diff
@@ -419,42 +392,34 @@ def test_expand_glob_join(self):
...
         else:
             # unchanged
             tests = (('/home/foo', '/home/foo'),)
-            tests_undo = tests
-        for test in tests:
-            self.assertEqual(extended_path(test[0]), test[1])
-        for test in tests_undo:
-            self.assertEqual(extended_path_undo(test[0]), test[1])
+        for short, extended in tests:
+            # already extended path shouldn't be changed
```
Comment (az0) [inline tests/TestFileUtilities.py]: These are logical tests

**Comment by az0** [comment]
This is great work. I love a cleanup like this that reduces code. 

I could merge this now, but would you rather wait? I see only 2/4 items are checked off.

**Comment by ROCKNROLLKID** [comment]
I am seeing merge conflicts with this. Will need to fix that first.

**Comment by tstenner** [comment]
I'd rather go through the last remaining files first and then rebase the PR, after that I'd also be in favor of merging it (less conflicts for me and everyone else). I can always open another PR for the remaining items.

**Comment by ROCKNROLLKID** [comment]
Are we holding back Python 3 coding for the next update then?

**Comment by az0** [comment]
@tstenner  OK, please write a note when it's ready.

@ROCKNROLLKID I would like to hold major changes including Python 3 and GTK 3 until after BleachBit 2.0 is released. Then it may also take a follow-up maintenance release to fix important regressions. After that I would like to evaluate how well Python 3 and GTK 3 work, especially on Windows. Given the recent work to shrink the new GTK 2 on Windows for BleachBit 2.0, I assume that I can also shrink Python 3 and GTK 3. This pull request is a stepping stone for Python 3 because it improves test coverage.

**Comment by tstenner** [comment]
The test for locked files fails because the worker reports that it cleaned too many bytes (6 instead of 3), but I (think I) didn't change anything relevant in the failing test. Could someone with a windows installation take a look at it?

**Comment by az0** [comment]
The error 6 != 3 looks like something I broke related to the new WindowsWipe code. I will work on it.

**Comment by az0** [comment]
The 6 != 3  error was fixed in commit 193e07a

**Comment by tstenner** [comment]
The unit tests pass now, travis hits a time limit somehow (downloading the appveyor artifacts?). We should merge this, I'll tackle the rest of the todo-list in a separate PR

**Comment by az0** [comment]
Yes, Travis hit a time limit waiting on AppVeyor, while Appveyor failed because of the errno typo.

**Pull request merged**